Editor: support multiframe and multiloop selection

ivan-mogilko opened the pull request:
Resolves #2124.

This adds support for the multi-frame selection to the View editor. This includes *selecting frames in different loops*.
When multiple frames are selected, the Property Grid will display shared values for them, and let set values for all frames at once (this is an existing feature, previously only utilized for Sprites).

Emulates classic selection controls:
* Click - reset selection to the given frame;
* Ctrl + Click - add or remove selection on the given frame;
* Shift + Click - select a range of frames, starting from the last *individually* selected frame and to the clicked one.

Ranged selection is supposed to work across loops too.

In terms of code, I added enum called MultiSelectAction that describes selection type, because I don't know any better. If there's an existing equivalent to this in C#, the I may replace this.

ivan-mogilko commented:
Something is wrong with CI not building windows part, so here's a manually built editor for testing:
https://www.dropbox.com/scl/fi/b3h3w3oxdwhfe4cuve5xr/361-multiviewframeselect.zip?rlkey=hckjbxjfozq7fsn7nd9xzm8qg&dl=0

PacketLauncher commented:
A great improvement to the View editor. Thanks! I can confirm that delaying , flipping, setting different images and setting a sound on frames in multiple loops work great! Below some feedback for other things I checked that don't currently work correctly:

1. Upon trying to delete (Del Key) a selection of frames in multiple loops, only the frames of the first loop are getting deleted while the selected frames on different loops remain intact.

2. The right-click context menu on any of the frames from the multiple selection removes the multiple selection and affects only the single frame clicked upon.

ivan-mogilko commented:
Updated, fixed selection reset on right click, made separate set of context commands depending on number of selected frames and whether they are selected in multiple loops.
